src/core/ext/client_channel/client_channel_factory.c

```diff
@@ -85,3 +90,120 @@ grpc_arg grpc_client_channel_factory_create_channel_arg(
   arg.value.pointer.vtable = &factory_arg_vtable;
   return arg;
 }
+
+struct grpc_channel_targets_info {
```

Comment:
Note that in #8842 I'm renaming mdstr_hash_table to slice_hash_table (because mdstr isn't a thing anymore).